.gitlab-ci.yml

```diff
@@ -39,18 +206,120 @@ nix-dry:
     - nix-build -v -v --dry-run ./release.nix --attr package.macos.x64.macho
 
 nix:
+  image: registry.gitlab.com/matrixai/engineering/maintenance/gitlab-runner
   stage: build
   interruptible: true
   script:
     - >
-        nix-build ./release.nix
-        --max-jobs $(nproc)
-        --attr application
-        --attr docker
+      BUILDOUT=`nix-build ./release.nix\
+        --attr application\
+        --attr docker\
+        --attr package.linux.x64.elf\
+        --attr package.windows.x64.exe\
+        --attr package.macos.x64.macho`;
+        mkdir builds;
+        cp -r $BUILDOUT builds/;
+        chmod -R a+rw builds;
   only:
     - master
+  artifacts:
+    paths:
+      - builds
+
+runs in node:
+  image: registry.gitlab.com/matrixai/engineering/maintenance/gitlab-runner
+  stage: quality
+  only:
+    - master
+  interruptible: true
+  script:
+    - >
+      nix-shell -I nixpkgs=./pkgs.nix --packages nodejs --run '
+      npm install;
+      echo password > testPassword;
+      npm run polykey -- agent start --test -np ./nodePath --password-file ./testPassword;
+      '
+
+running application build:
+  image: registry.gitlab.com/matrixai/engineering/maintenance/gitlab-runner
+  stage: quality
+  only:
+    - master
+  needs:
+    - job: nix
+      artifacts: true
+  interruptible: true
+  script:
+    - >
+      nix-shell -I nixpkgs=./pkgs.nix --packages nodejs --run '
+      echo password > testPassword;
+      node builds/*/lib/node_modules/@matrixai/polykey/dist/bin/polykey.js agent start -np ./nodePath --password-file ./testPassword --test;
+      '
+
+
+running docker build:
+  image: docker:19.03.12
+  variables:
+    DOCKER_TLS_CERTDIR: "/certs"
+  services:
+    - docker:19.03.12-dind
```

Comment:
Should review this in relation to https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#use-the-docker-executor-with-the-docker-image-docker-in-docker

Reply:
In https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#docker-in-docker-with-tls-enabled step 3 is the example I used for this.